django/utils/timesince.py

```diff
@@ -69,22 +71,37 @@ def timesince(d, now=None, reversed=False, time_strings=None, depth=2):
             leapdays += 1
     delta -= datetime.timedelta(leapdays)
 
+    # adjust months duration to be relative to the months between d and now
+    months = now.month - d.month
+    if months > 0:
+        m_days = (sum(MONTHS_DAYS[d.month - 1 : now.month - 1])) / months
+    else:
+        months = 12 + months  # note: we also need this later to fix edge cases.
+        m_days = (365 - sum(MONTHS_DAYS[now.month - 1 : d.month - 1])) / months
+    time_chunks = list(TIMESINCE_CHUNKS)
+    time_chunks[1] = (int(m_days * 24 * 60 * 60), "month")
+
     # ignore microseconds
     since = delta.days * 24 * 60 * 60 + delta.seconds
     if since <= 0:
         # d is in the future compared to now, stop processing.
         return avoid_wrapping(time_strings["minute"] % {"num": 0})
-    for i, (seconds, name) in enumerate(TIMESINCE_CHUNKS):
+    for i, (seconds, name) in enumerate(time_chunks):
         count = since // seconds
         if count != 0:
             break
     else:
         return avoid_wrapping(time_strings["minute"] % {"num": 0})
+
     result = []
     current_depth = 0
-    while i < len(TIMESINCE_CHUNKS) and current_depth < depth:
-        seconds, name = TIMESINCE_CHUNKS[i]
+
+    while i < len(time_chunks) and current_depth < depth:
+        seconds, name = time_chunks[i]
         count = since // seconds
+        if name == "month":
+            # count might overshoot if now.month is longer than the month average length
```

Comment:
@GianpaoloBranca Can I just clarify, did you mean `if now.month is shorter than the month average`?